### PR TITLE
add golang-1.18 spec

### PR DESCRIPTION
**Pull request by nicogbg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

add golang-1.18 spec

###### Change Log  

- Change
- Change
- Change

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: SPECS/golang/golang-1.18.spec [hunk 1]
```diff
@@ -0,0 +1,247 @@
+%global goroot          %{_libdir}/golang
```
Comment (mfrw) [inline SPECS/golang/golang-1.18.spec:1]: Should we not add a `version` constraint on the offending packages.
e.g
`k3s, helm, terraform` ?

File: SPECS/golang/golang-1.18.spec [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+Distribution:   Mariner
+Group:          System Environment/Security
+URL:            https://golang.org
+Source0:        https://golang.org/dl/go%{version}.src.tar.gz
+Source1:        https://dl.google.com/go/go1.4-bootstrap-20171003.tar.gz
+Patch0:         go14_bootstrap_aarch64.patch
+# CVE-2022-41717 is fixed in 1.18.9
+Patch1:         CVE-2022-41717.patch
+Obsoletes:      %{name} < %{version}
+Provides:       %{name} = %{version}
```
Comment (PawelWMS) [inline SPECS/golang/golang-1.18.spec:28]: Isn't this the default behaviour for all packages? Or is one of the macro (`_use_internal_dependency_generator 0`?) messing with RPM's automation and thus this is required?

**Review by PawelWMS — APPROVED** [review]

File: SPECS/golang/golang-1.18.spec (hunk 1 above)
Comment (nicogbg) [reply SPECS/golang/golang-1.18.spec:1]: it is already done in the relevant spec. I assumed that full build would get golang 1.18.8 from PMC but it do not
so I created golang 1.18 spec

File: SPECS/golang/golang-1.18.spec (hunk 2 above)
Comment (nicogbg) [reply SPECS/golang/golang-1.18.spec:28]: I've taken the golang spec as it was before moving to 1.19 and just created a spec for version 1.18 from it (nothing changed)

**Pull request merged**